syntax = "proto2";

import "aos/vision/image/camera_params.proto";

package y2017.vision;

message GameSpecific {
  // Needs more woojy.
  optional int32 woojy = 1;
}

// Configurations that may be robot dependent.
message RobotConfig {
  // Device name of the camera (ie /dev/video1).
  optional string camera_device_path = 1;

  // RoboRIO IP address.
  optional string roborio_ipaddr = 2;

  // Jetson board IP address.
  optional string jetson_ipaddr = 3;

  // Port to use (both sides).
  optional int32 port = 4;
}

// Stores configuration information for a given set of hardware.
message VisionConfig {
  // Map robot name to the robot dependent configuration.
  map<string, RobotConfig> robot_configs = 1;

  // Year independent camera parameters.
  optional aos.vision.CameraParams camera_params = 3;

  // Parameters for this specific game
  optional GameSpecific game_params = 4;
}
